datamade/ccfp-asset-dashboard
# added logic to the total_score method + tests for the method

## Overview
This PR has 
- a model method for totaling the score for a project. The calculations are based on a project's ProjectScore and the ScoreWeights.
- tests for the method
- a UI change to show the total score on the project detail page
- a DB migration that missed out on PR #24...

It's for issue #32.

### Demo
![Screen Shot 2021-02-04 at 3 05 47 PM](https://user-images.githubusercontent.com/38969506/106957840-25647900-66fe-11eb-80bb-d6e44a9f5402.png)

### Notes
See my note in the issue: https://github.com/datamade/ccfp-asset-dashboard/issues/32#issuecomment-773553971. 

## Testing Instructions
- Visit the review app and see the score on a project detail page.
- Change the score in the UI. When you save, it should be updated.

- Pull down the code locally and run the tests: `docker-compose -f docker-compose.yml -f tests/docker-compose.yml run --rm app`

- You can also test in the console. You'll first need to load the fixtures so the score weight exists: `docker-compose run --rm app python manage.py loaddata asset_dashboard/fixtures/data.json`
- Here is some code I was copy/pasting into the app's python shell while writing this method:
```python
from asset_dashboard.models import ScoreWeights, Project, ProjectScore
project = Project(name="new trailhead", description="we need a new trailhead")
new_score = ProjectScore(project=project, core_mission_score=3, operations_impact_score=4, sustainability_score=2, ease_score=5, geographic_distance_score=2, social_equity_score=4)
new_score.total_score
# prints 14.3
```
- Check the math
